halo2_proofs/src/plonk/keygen.rs

```diff
@@ -349,3 +349,118 @@ where
         ev,
     })
 }
+
+/// Generate a `ProvingKey` from a `VerifyingKey` and an instance of `Circuit`.
+pub(crate) fn keygen_pk_from_info <C>(
+    params: &Params<C>,
+    vk: &VerifyingKey<C>,
+    fixed: Vec<Polynomial<C::Scalar, LagrangeCoeff>>,
+    permutation: permutation::keygen::Assembly,
+) -> Result<ProvingKey<C>, Error>
+where
+    C: CurveAffine,
+{
+    let cs = vk.cs.clone();
+    let selectors = vec![vec![false; params.n as usize]; cs.num_selectors];
```

Comment:
We'd better to assert selector number == 0. It cannot handle selector correctly.